### DISCUSSION TITLE
Are there plans to support auto-converge?

### DISCUSSION THREAD

**Discussion by Lxiaoyouling** [cloud-hypervisor/cloud-hypervisor#5284] [General]
Hello,

Does cloud-hypervisor have plans to support auto-converge  like in qemu？
Because live migration downtime with cloud-hypervisor is too long when memory writes are frequent.

**Comment by likebreath** [comment]
Thanks for sharing your experiences with Cloud Hypervisor. Our current implementation of live migration has a fixed number of iterations (e.g. 5) for migrating dirt pages of guest memory [1]. It was picked mostly for the sake of simplicity. 

So I definitely agree that this is a great area for optimization. I personally haven't looked into the auto-converge algorithm you mentioned, so can't comment much the algorithm. I would suggest that you open a feature request and explain why you think this is a good optimization to add for Cloud Hypervisor. 

[1] https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/vmm/src/lib.rs#L1572-L1579
